File: archive/r/rust/longest-word.rs
use std::env::args;
use std::process::exit;

fn usage() -> ! {
    println!("Usage: please provide a string");
    exit(0);
}

fn get_longest_word_len(s: &str) -> usize {
    s.trim()
        .split_whitespace()
        .map(|t| t.len())
        .max()
        .unwrap()
}

fn main() {
    let mut args = args().skip(1);

    // Exit if 1st command-line argument is empty
    let s: &str = &args
        .next()
        .unwrap_or_else(|| usage());
    if s.len() < 1 {
        usage();
    }

    // Get longest word length and display it
    println!("{}", get_longest_word_len(s));
}

File: archive/c/c/longest-word.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

void error(){
    printf("Usage: please provide a string\n");
    exit(0);
}
int main(int argc, char *argv[])
{
    if(argc<2){
        error();
    }
    if(strlen(argv[1])<=0){
        error();
    }
    int max = -1;
    char* word = strtok(argv[1], " ,\n\t");
    while (word != NULL) {
        int len = strlen(word);
        if(len>max){
            max = len;
        }
        word = strtok(NULL, " ,\n\t");
    }

    printf("%d",max);
      
    
    return 0;
}
